### PR TITLE
Add account level security group configuration options

### PR DESCRIPTION
**Pull request by sijis**
These changes allow to specific security groups for elb and ec2 instances per account.

Currently the only option is to set sg for all account. This gives a little more flexibility on what sg are added to a specific account.

### CODE REVIEW

File: src/foremast/consts.py [hunk 1]
```diff
@@ -139,6 +141,53 @@ def find_config():
...
+            entries.pop(count)
+    return entries
+
+
+def _convert_string_to_native(value):
+    """Convert a string to its native python type"""
+    result = None
+
+    try:
+        result = ast.literal_eval(value)
```
Comment (e4r7hbug) [inline src/foremast/consts.py]: I don't know if we need `ast` here. If we're possibly expecting JSON, maybe we should use `json`? A test case for this sole function would help clarify what the incoming data is.

File: src/foremast/consts.py [hunk 2]
```diff
@@ -139,6 +141,53 @@ def find_config():
...
+
+
+def _generate_security_groups(config_key):
+    """Read config file and generate security group dict by environemnt
+
+    Args:
+        config_key (str): Configuration file key
+
+    Returns:
+        dict: of environments in {$env: [$group1, group2]} format
```
Comment (e4r7hbug) [inline src/foremast/consts.py]: Probably don't need the `$` in the example, looks fine without.

File: src/foremast/consts.py [hunk 3]
```diff
@@ -139,6 +141,53 @@ def find_config():
...
+    try:
+        result = ast.literal_eval(value)
+    except (SyntaxError, ValueError):
+        # Likely a string
+        result = value.split(',')
+    return result
+
+
+def _generate_security_groups(config_key):
+    """Read config file and generate security group dict by environemnt
```
Comment (e4r7hbug) [inline src/foremast/consts.py]: Spell check and period.

File: src/foremast/consts.py (hunk 1 above)
Comment (sijis) [inline src/foremast/consts.py]: It could be a dict too, not just json.

It needs to handle a `config.py` too.

Comment (e4r7hbug) [inline src/foremast/consts.py]: Bwah.

File: src/foremast/consts.py [hunk 4]
```diff
@@ -140,6 +142,54 @@ def find_config():
     return configurations
 
 
+def _remove_empty_entries(entries):
+    """Remove empty entries in a list"""
+    valid_entries = []
+    for entry in entries:
```
Comment (e4r7hbug) [inline src/foremast/consts.py]: Minor optimization would be `set(entries)`. Not a real concern, but something to note.

**Comment by e4r7hbug** [comment]
Looks good for me.

File: src/foremast/consts.py (hunk 4 above)
Comment (sijis) [inline src/foremast/consts.py]: addressed

**Pull request merged**